### PR TITLE
[Feature] Replace the implementation of deform_roi_pool with mlu-ops

### PR DESCRIPTION
**Pull request by defei-coder**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Replace the implementation of deform_roi_pool with mlu-ops.

## Modification

1. Delete mmcv/ops/csrc/common/mlu/deform_roi_pool_mlu_kernel.mlu.
2. Modify mmcv/ops/csrc/pytorch/mlu/deform_roi_pool_mlu.cpp.
3. Modify mmcv/ops/csrc/pytorch/mlu/mlu_common_helper.cpp.
4. Modify mmcv/ops/csrc/pytorch/mlu/mlu_common_helper.h.
5. Modify setup.py.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/ops/csrc/pytorch/mlu/mlu_common_helper.cpp [hunk 1]
```diff
@@ -72,6 +72,39 @@ void MluOpTensorDescriptor::set(Tensor t) {
...
+                                            mluOpTensorLayout_t layout) {
+  mluOpDataType_t data_type = getMluOpDataType(t.dtype());
+  int t_dim = t.dim();
+  std::vector<int> shape_info = checkUpperBoundAndCastTo<int>(t.sizes().vec());
+  std::vector<int> stride_info =
+      checkUpperBoundAndCastTo<int>(t.strides().vec());
+  if (layout == CNNL_LAYOUT_NHWC || layout == CNNL_LAYOUT_NDHWC ||
+      layout == CNNL_LAYOUT_NLC) {
+    convertShapeAndStride(shape_info, stride_info);
+  } else {
```
Comment (Qiza-lyhm) [inline mmcv/ops/csrc/pytorch/mlu/mlu_common_helper.cpp:85]: 这里需要判断if layout==HWCN么？

File: mmcv/ops/csrc/pytorch/mlu/mlu_common_helper.cpp [hunk 2]
```diff
@@ -72,6 +72,39 @@ void MluOpTensorDescriptor::set(Tensor t) {
...
 }
 
+void MluOpTensorDescriptor::set_with_layout(Tensor t,
+                                            mluOpTensorLayout_t layout) {
+  mluOpDataType_t data_type = getMluOpDataType(t.dtype());
+  int t_dim = t.dim();
+  std::vector<int> shape_info = checkUpperBoundAndCastTo<int>(t.sizes().vec());
+  std::vector<int> stride_info =
+      checkUpperBoundAndCastTo<int>(t.strides().vec());
+  if (layout == CNNL_LAYOUT_NHWC || layout == CNNL_LAYOUT_NDHWC ||
```
Comment (WangYuan-cn) [inline mmcv/ops/csrc/pytorch/mlu/mlu_common_helper.cpp:82]: MLUOP_*

File: mmcv/ops/csrc/pytorch/mlu/mlu_common_helper.cpp [hunk 3]
```diff
@@ -72,6 +72,39 @@ void MluOpTensorDescriptor::set(Tensor t) {
   set_desc(t, layout, data_type, dim_array);
 }
 
+void MluOpTensorDescriptor::set_with_layout(Tensor t,
```
Comment (WangYuan-cn) [inline mmcv/ops/csrc/pytorch/mlu/mlu_common_helper.cpp:75]:
```suggestion
void MluOpTensorDescriptor::set_layout(Tensor t,
```

File: mmcv/ops/csrc/pytorch/mlu/mlu_common_helper.h [hunk 4]
```diff
@@ -52,3 +53,47 @@ class MluOpHandle {
...
+void convertShapeAndStride(std::vector<T>& shape_info,
+                           std::vector<T>& stride_info) {
+  TORCH_MLU_CHECK(shape_info.size() == stride_info.size(),
+                  "shape size need equal to stride size.");
+  const int dim = shape_info.size();
+  std::vector<T> temp_shape_info(dim);
+  std::vector<T> temp_stride_info(dim);
+  temp_shape_info[0] = shape_info[0];
+  temp_stride_info[0] = stride_info[0];
+  for (size_t i = 0; i < dim - 1; ++i) {
```
Comment (WangYuan-cn) [inline mmcv/ops/csrc/pytorch/mlu/mlu_common_helper.h:74]:
```suggestion
  for (int i = 0; i < dim - 1; ++i) {
```

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**